File: AndroidStudioProjects/DaggerDemoJava/app/src/main/java/com/example/myapplication/Car.java
package com.example.myapplication;

import android.util.Log;

import javax.inject.Inject;

public class Car {
    private static final String TAG = "Car";

    private Engine engine;
    private Wheels wheels;
    private Driver driver;

    @Inject
    public Car(Driver driver,Engine engine, Wheels wheels) {
        Log.d(TAG, "Car: Constructor");
        this.driver= driver;
        this.engine = engine;
        this.wheels = wheels;
    }

    @Inject
    public void passCarToRemote(Remote remote){
        Log.d(TAG, "passCarToRemote: ");
        remote.setCar(this);
    }

    public void start(){
        engine.start();
        Log.d(TAG, "DRIVER: "+driver);
        Log.d(TAG, "driving....");
    }
}

File: AndroidStudioProjects/DaggerDemoKotlin/app/src/main/java/com/example/daggerdemokotlin/Car.kt
package com.example.daggerdemokotlin

import javax.inject.Inject

class Car {
    @Inject
    lateinit var wheels: Wheels
    lateinit var engine: Engine
    lateinit var driver: Driver

    @Inject
    constructor(engine: Engine,driver: Driver){
        this.engine= engine
        this.driver= driver
    }

    @Inject
    fun provideCarToRemote(remote: Remote){
        remote.provideCar(this)
    }

    fun start(){
        System.out.println("Driver: "+driver)
        //System.out.println("wheels: "+wheels)
        //engine.start()
        //System.out.println("driving.....")
    }
}